src/client/datascience/jupyter/kernels/types.ts

```diff
@@ -112,6 +113,14 @@ export function isLocalConnection(
     );
 }
 
+export function isLocalHostConnection(kernelConnection: KernelConnectionMetadata): boolean {
+    if (kernelConnection.kind === 'connectToLiveKernel' || kernelConnection.kind === 'startUsingRemoteKernelSpec') {
+        const parsed = new url.URL(kernelConnection.baseUrl);
+        return parsed.hostname.toLocaleLowerCase() === 'localhost' || parsed.host === '127.0.0.1';
```

Comment:
I was going to ask if localhost is more configurable, and I believe it is. But I think that we only have to worry about how Jupyter starts it, so this is fine I believe.

Reply:
Was going to ask if we needed to check for an npm module or something for a more comprehensive loopback check, but I don't think we need to.